### ISSUE TITLE
Add coupled MCMC methods

### ISSUE THREAD

**Issue by AdrienCorenflos** [blackjax-devs/blackjax#64]
Ideally implement these two different papers

- https://arxiv.org/abs/1709.00404
- https://arxiv.org/abs/2104.05134

They both build on this https://rss.onlinelibrary.wiley.com/doi/full/10.1111/rssb.12336 and can be used to compute unbiased estimates of expectations. I don't really thing these methods are implemented in any Python library (I don't think it's in Stan either?) and would be a cool show case.

**Comment by rlouf** [comment]
Really cool ! You shouldn't have to reimplement the HMC logic for the simplest coupled algorithms, let me know if you find yourself doing that.

**Comment by rlouf** [comment]
Very close to #38. There I think the authors run two long coupled chains and keep all samples; here if i understand well we run many coupled chains and only keep the last sample. Correct me if I'm wrong.

**Comment by AdrienCorenflos** [comment]
What do you mean by "we only keep the last sample"? Of the chain or of the
integrator? 'Cause we keep the whole chains for sure.

**Comment by rlouf** [comment]
You're right. I'll need to re-read the paper, afair the difference is that in Heng & Jacob's paper they want "to obtain estimators that are consistent in the limit of the number of replicates" while in the paper referenced in #38 they run two coupled chains for a long time.

But it does not matter here !

**Comment by AdrienCorenflos** [comment]
Well, this swindled algo will require to open up the method to pass in
antithetic moments, so I would suggest keeping it the way I have put it if
you are planning on implementing it (although I agree that for my purpose I
could simply have run two chains with CRNs, albeit at the price of some
efficiency I presume).

**Comment by rlouf** [comment]
> Well, this swindled algo will require to open up the method to pass in
> antithetic moments, 

We thought about it a while ago with @junpenglao and if I remember well the solution is to allow to pass the direction to the kernel.

**Comment by AdrienCorenflos** [comment]
Isn't this complexifying the interface quite a bit though?

**Comment by rlouf** [comment]
Not so much since you can _configure_ the kernel to integrated in a given direction in time when building it.

**Comment by rlouf** [comment]
CC @pierrejacob

**Comment by rlouf** [comment]
https://arxiv.org/abs/2104.05134

**Comment by rlouf** [comment]
https://arxiv.org/abs/2206.07202
